src/main/java/io/github/in_toto/models/Metablock.java

```diff
@@ -80,7 +85,27 @@ public void dump(Writer writer)
      * @return a JSON string representation of the metadata instance
      */
     public String dumpString() {
-        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        Gson gson = new GsonBuilder()
+                // Override gson serializer to convert all numbers to
+                // non-floating point numbers, as we don't allow floating point
+                // numbers in the reference implementation.
+                // This is required to handle generic data, such as
+                // `byproducts` or `environment`, where we don't declare the
+                // type of the contained values. Gson treats any numeric value
+                // where the type is not specified in the target data structure
+                // as double.
+                .registerTypeAdapter(Double.class, new JsonSerializer<Double>() {
```

Comment:
I know this is "correct" Java, but i personally hate inline class declarations. can we give this a reasonable name? e.g., `ReasonableJsonDoubleSerializer` and declare it as a nested static(?) class or so?

Reply:
Note, my e.g., is a **joke**

Reply:
Haha, sure, will do.